src/Microsoft.Azure.SignalR.Management/IInternalServiceHubContext.cs

```diff
@@ -16,14 +13,8 @@ internal interface IInternalServiceHubContext : IServiceHubContext
         /// <summary>
         /// Gets client endpoint access information object for SignalR hub connections to connect to Azure SignalR Service
         /// </summary>
-        /// <param name="httpContext">The HTTP context which might provide information for routing and generating access token.</param>
-        /// <param name="userId">The user ID. If null, the identity name in <see cref="HttpContext.User" /> of <paramref name="httpContext"/> will be used.</param>
-        /// <param name="claims">The claim list to be put into access token. If null, the claims in <see cref="HttpContext.User"/> of <paramref name="httpContext"/> will be used.</param>
-        /// <param name="lifetime">The lifetime of the token. The default value is one hour.</param>
-        /// <param name="isDiagnosticClient">The flag whether the client to be connected is a diagnostic client.</param>
-        /// <param name="cancellationToken">Cancellation token for aborting the operation. If null, the <see cref="HttpContext.RequestAborted"/> of <paramref name="httpContext"/> will be used. </param>
         /// <returns>Client endpoint and access token to Azure SignalR Service.</returns>
-        Task<NegotiationResponse> NegotiateAsync(HttpContext httpContext = null, string userId = null, IList<Claim> claims = null, TimeSpan? lifetime = null, bool isDiagnosticClient = false, CancellationToken cancellationToken = default);
+        Task<NegotiationResponse> NegotiateAsync(NegotiationOptions options = null, CancellationToken cancellationToken = default);
```

Comment:
not necessary to allow `null` options, add a defense check is enough?

Reply:
https://github.com/Azure/azure-signalr/blob/b54a50f41486d4ca590f6ce2b8ecdc72cfee1501/src/Microsoft.Azure.SignalR.Management/ServiceManager.cs#L63 
`ServiceManager` allows `userId` to be null, so the scene with no parameter is valid.